cmd/testwrapper/testwrapper.go

```diff
@@ -135,33 +142,43 @@ func runTests(ctx context.Context, attempt int, pt *packageTests, otherArgs []st
 			// ignore
 		case "run":
 			resultMap[name] = &testAttempt{
-				name: name,
+				Name:         name,
+				AttemptCount: attempt,
 			}
 		case "skip", "pass", "fail":
-			resultMap[name].outcome = goOutput.Action
+			resultMap[name].Outcome = goOutput.Action
 			ch <- resultMap[name]
 		case "output":
 			if strings.TrimSpace(goOutput.Output) == flakytest.FlakyTestLogMessage {
-				resultMap[name].isMarkedFlaky = true
+				resultMap[name].IsMarkedFlaky = true
 			} else {
 				resultMap[name].logs.WriteString(goOutput.Output)
 			}
 		}
 	}
+	if opt.w != nil {
+		for _, result := range resultMap {
+			testAttemptJson, _ := json.Marshal(result)
+			_, err = opt.w.Write(testAttemptJson)
+			if err != nil {
+				log.Printf("error appending to test attempt json file: %v", err)
+			}
+		}
+	}
 	<-done
 }
 
 func main() {
 	ctx := context.Background()
 
-	// We only need to parse the -v flag to figure out whether to print the logs
-	// for a test. We don't need to parse any other flags, so we just use the
-	// flag package to parse the -v flag and then pass the rest of the args
-	// through to 'go test'.
+	// We need to parse the -v flag to figure out whether to print the logs
+	// for a test.
+	// The -w flag is to indicate whether we want to write the json test results to a file.
 	// We run `go test -json` which returns the same information as `go test -v`,
 	// but in a machine-readable format. So this flag is only for testwrapper's
 	// output.
 	v := flag.Bool("v", false, "verbose")
+	w := flag.Bool("w", false, "write")
```

Comment:
this needs more docs. `write` what? 

Have it be an explicit string flag instead which takes in the filename

```
jsonFile := flag.String("json", "", "if set, the file to store all test results in")
```

`--json=test_attempts.json`